refactor(core): move linkedSignal implementation to primitives

pkozlowski-opensource opened the pull request:
This change refactors the Angular-specific linkedSignal implementation such that its core logic can be shared with other frameworks.

thePunderWoman reviewed (approved):
reviewed-for: fw-general, public-api, primitives

JeanMeche commented on packages/core/primitives/signals/src/linked_signal.ts:
```diff
@@ -0,0 +1,184 @@
+    producerRecomputeValue(node: LinkedSignalNode<unknown, unknown>): void {
+      if (node.value === COMPUTING) {
+        // Our computation somehow led to a cyclic read of itself.
+        throw new Error('Detected cycle in computations.');
```
nit: Can't we maybe share this string with the one in `packages/core/primitives/signals/src/computed.ts` ?

pkozlowski-opensource replied:
I want to do a bit more merging of code between computed and linked signal in a follow up, will take care of the error message then

alxhub commented on packages/core/primitives/signals/src/linked_signal.ts:
```diff
@@ -0,0 +1,184 @@
+ * A dedicated symbol used before a state value has been set / calculated for the first time.
+ * Explicitly typed as `any` so we can use it as signal's value.
+ */
+const UNSET: any = /* @__PURE__ */ Symbol('UNSET');
```
We could share these symbols with `computed` rather than redefining them here.

pkozlowski-opensource replied:
👍 done

pkozlowski-opensource committed:
refactor(core): move linkedSignal implementation to primitives

This change refactors the Angular-specific linkedSignal implementation
such that its core logic can be shared with other frameworks.

e-cline reviewed (approved):
Looks good for primitives-shared and Wiz - Thank you!

Reviewed-for: primitives-shared

alxhub reviewed (approved):
Reviewed-for: public-api
Reviewed-for: primitives

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

pkozlowski-opensource commented:
Note: will run a TGP on it on Mon

AndrewKushnir commented:
This PR was merged into the repository by commit 04c24472d29673522a9b13f42dfe90796266cafc.

The changes were merged into the following branches: main, 19.1.x
